tests/conftest.py

```diff
@@ -83,6 +94,97 @@ def incomplete_application():
     return application
 
 
+@pytest.fixture
+def build_release(build_release_path):
+    def func(names, **kwargs):
+        requested_files = [{"name": n} for n in names]
+        release = ReleaseFactory(requested_files=requested_files, **kwargs)
+
+        build_release_path(release)
+
+        return release
+
+    return func
+
+
+@pytest.fixture
+def build_release_with_files(build_release, build_release_file):
+    """Build a Release and generate some files for the given names"""
+
+    def func(names, **kwargs):
+        release = build_release(names, **kwargs)
+
+        for name in names:
+            build_release_file(release, name)
+
+        return release
+
+    return func
+
+
+@pytest.fixture
+def build_release_file(file_content):
+    """
+    Build a ReleaseFile
+
+    Given a Release instance and a filename create both the ReleaseFile object
+    and the on-disk file with random content from the file_content fixture.
+    """
+
+    def func(release, name):
+        # build a relative path for the file
+        path = Path(release.workspace.name) / "releases" / str(release.id) / name
+
+        rfile = ReleaseFileFactory(
+            release=release,
+            workspace=release.workspace,
+            name=name,
+            path=path,
+            filehash=hashlib.sha256(file_content).hexdigest(),
+            size=len(file_content),
+            uploaded_at=timezone.now(),
+        )
+
+        # write the file to disk
+        rfile.absolute_path().write_bytes(file_content)
+
+        return rfile
+
+    return func
+
+
+@pytest.fixture
+def build_release_path(tmp_path):
```

Comment:
Still reading through (had a few meetings, have one more), but I was wondering if `build_release_path` and `build_release_file` needed to be fixtures? They're only used in conftest, so they feel like utility functions?

Reply:
They don't need to be fixtures but I exposed them in case they were of use in the future.  `content_file` for example ended up being really useful for example.